tianshou/data/stats.py

```diff
@@ -56,6 +56,8 @@ class InfoStats(DataclassPPrintMixin):
 
     gradient_step: int
     """The total gradient step."""
+    best_score: float
```

Comment:
Rename to `best_custom_score`? Should be optional and the docstring should be extended.

Overall, I'm a bit skeptical about adding it here, since the semantics of it is not well defined and depends on things configured in other places. An alternative to a fixed name would be to extend the `InfoStats` with 

```
metadata: dict = field(default_factory=dict)
```

but this won't work with current Trainer structure.


Generally, looking at the trainer design, the termination of the training should either follow a strategy pattern or take a callable that calculates the score based on InfoStats. Termination should also be based on InfoStats, or the whole training. That would be too much of a change, so maybe we can merge this now and make a larger breaking change later.

@opcode81 wdyt?

Reply:
After another look, I guess it's ok to have scores decoupled from rewards as a central concept, so it can be left as is. Resolving